sound/soc/sof/ipc.c

```diff
@@ -906,7 +906,14 @@ int sof_ipc_init_msg_memory(struct snd_sof_dev *sdev)
 	struct snd_sof_ipc_msg *msg;
 
 	msg = &sdev->ipc->msg;
-	msg->msg_data = devm_kzalloc(sdev->dev, SOF_IPC_MSG_MAX_SIZE, GFP_KERNEL);
+
+	if (SOF_ABI_VERSION_MAJOR(sdev->fw_ready.version.abi_version) >= 4)
+		msg->msg_data_size = SOF_IPC4_MSG_MAX_SIZE;
+	else
+		msg->msg_data_size = SOF_IPC_MSG_MAX_SIZE;
+
+	msg->msg_data = devm_kzalloc(sdev->dev, msg->msg_data_size,
+				     GFP_KERNEL);
```

Comment:
Should not we have the `msg->reply_data` allocated with the same size?

Reply:
and while there, can we rename the `msg_data_size` to `max_payload_size` or something to make it less confusing when reading code?

Reply:
@bardliao  sof_ipc_init_msg_memory is called by sof_fw_ready.  I made such change and want to introduce another function sof_ipc4_init_msg_memory called by sof_ipc4_fw_ready, so you can set the msg size in that function.

Reply:
> Should not we have the `msg->reply_data` allocated with the same size?

Will do.

> and while there, can we rename the msg_data_size to max_payload_size or something to make it less confusing 
when reading code?

msg_data_size means the size of msg_data. Is it confusing?

> @bardliao sof_ipc_init_msg_memory is called by sof_fw_ready. I made such change and want to introduce another function sof_ipc4_init_msg_memory called by sof_ipc4_fw_ready, so you can set the msg size in that function.

Will do.

Reply:
> msg_data_size means the size of msg_data. Is it confusing?
@bardliao, yes it is.

The `struct snd_sof_ipc_msg` have msg_data, msg_size, reply_data, reply_size and now introducing the msg_data_size which is the allocated size behind of msg_data and reply_data.
It would be less confusing if it is named after it's use, max_msg_size or max_payload_size as it applies to both msg and the reply sizes.

It is completely unrelated that the `struct snd_sof_ipc_msg` is really not a struct for IPC message but it is a struct for an IPC transfer, The message is behind of the msg_data.